GOVERNANCE.md

```diff
@@ -61,19 +61,67 @@ maintainer has already discussed the candidacy with the candidate and a
 maintainer is willing to be a sponsor by opening the pull request. The candidate
 becomes a maintainer once the pull request is merged.
 
-## Adding sub-projects
-
-Similar to adding maintainers, new sub projects can be added to containerd
-GitHub organization as long as they adhere to the CNCF
-[charter](https://www.cncf.io/about/charter/) and mission. After a project
-proposal has been announced on a public forum (GitHub issue or mailing list),
-the existing maintainers are  given five business days to discuss the new
+## Adding subprojects
+
+containerd subprojects are divided into two flavors currently: **core** and
+**non-core**. Most of the repositories within the containerd GitHub organization are
+"core" to the delivery of the containerd runtime project's releases. For example,
+the Golang implementations of the cgroups, runc, and console APIs, the ttrpc
+implementation, and various plugins like the CRI implementation. These, among a
+handful of other repositories, are all **core** and required pieces which culminate
+in officially supported and maintained releases of containerd.
+
+Non-core projects have strong affiliation with the containerd runtime, but
+operate in some ways similar to the traditional `contrib/` directory in many
+open source projects. They are most likely separately maintained, and potentially
+have a different release cycle and support guarantee than the main containerd project
+and its core subprojects. They may be written in other languages and therefore
+require different skills, developer tools, and CI systems than the core projects.
+For these reasons, non-core subprojects will be treated differently than core
+projects within the governance of the containerd organization.
+
+Both core and non-core subprojects must adhere to the CNCF
+[charter](https://www.cncf.io/about/charter/) and mission.
+
+### Adding core subprojects
+
+New core subprojects can request to be added to the containerd GitHub
+organization by submitting a project proposal via public forum (a
+`containerd/project` GitHub issue is the easiest way to provide this proposal).
+The existing maintainers are given five business days to discuss the new
 project, raise objections and cast their vote. Projects must be approved by at
 least 66% of the current maintainers by adding their vote.
 
 If a project is approved, a maintainer will add the project to the containerd
 GitHub organization, and make an announcement on a public forum.
 
+To make this transition to being a core subproject easier, please prepare your
+repository with the DCO signoff checks, the fileheader check, and by adding the
+appropriate Apache 2 license file to your project root as well as the boilerplate
+header to all applicable source files.
```

Comment:
> Most of the repositories within the containerd GitHub organization are "core" to the delivery of the containerd runtime project's releases.

Does `core` == `part of release`?

Does a new core project need to be integrated into containerd release? Either a built-in plugin, or a binary in the release tarball? And is that a prerequisite of becoming core?

Reply:
I don't think we should have integration requirements. I think being part of core means it is supported as part of a release though, being in the main release binary or tarballs will always make that true but we could foresee other cases. I think from the release logic is makes sense though since anyone releasing code that is part of a containerd release must be a "core" maintainer. So promoting a project to core would just be for that purpose and bringing in those maintainers as project level maintainers.

Reply:
I see.

A core project may have release artifact somewhere else, but the release is considered as part of containerd release schedule. SG. :)